app/bundles/LeadBundle/Entity/LeadRepository.php

```diff
@@ -1088,6 +1088,33 @@ public function getLeadOwner($ownerId)
         return $result;
     }
 
+    /**
+     * Check Lead segments by ids.
+     *
+     * @param array<integer> $stages
+     */
+    public function checkLeadStage(Lead $lead, array $stages = []): bool
+    {
+        if (empty($stages)) {
+            return false;
+        }
+
+        $q = $this->getEntityManager()->getConnection()->createQueryBuilder();
+        $q->select('s.id')
+            ->from(MAUTIC_TABLE_PREFIX.'stages', 's');
+        $q->join('s', MAUTIC_TABLE_PREFIX.'leads', 'l', 'l.stage_id = s.id')
+            ->where(
+                $q->expr()->andX(
+                    $q->expr()->in('s.id', ':stageIds'),
+                    $q->expr()->eq('l.id', ':leadId')
+                )
+            )
+            ->setParameter('stageIds', implode(',', $stages))
```

Comment:
This is more secure:
```suggestion
            ->setParameter('stageIds', $stages, \Doctrine\DBAL\Connection::PARAM_INT_ARRAY)
```